pydax/_dataset.py

```diff
@@ -80,105 +83,115 @@ def _data_dir(self) -> pathlib.Path:
         if not self._data_dir_.exists():
             self._data_dir_.mkdir(parents=True)
         elif not self._data_dir_.is_dir():  # self._data_dir_ exists and is not a directory
-            raise FileExistsError(f'"{self._data_dir_}" exists and is not a directory.')
+            raise NotADirectoryError(f'"{self._data_dir_}" exists and is not a directory.')
         return self._data_dir_
 
+    @property
+    def _pydax_dir_(self) -> pathlib.Path:
+        "Cache, metainfo, etc. directory used by this class."
+        return self._data_dir_ / '.pydax.dataset'
+
     @property
     def _pydax_dir(self) -> pathlib.Path:
```

Comment:
The naming here will get confusing later on for us. Maybe we can rename to something like `_return_pydax_dir` and `_create_and_return_pydax_dir`? That's not perfect either though.

Reply:
Which part do you think would become confusing? I think it's fine though. It's supposed to be used just like a read-only attribute.

Reply:
How about we think about this in a separate PR? We can resolve `_data_dir` as well.

Reply:
sure thing, we can do in a separate PR